frontend/src/App.tsx

```diff
@@ -40,30 +70,48 @@ function App() {
             <header className="App-header">
                 <h1>Event Hub - Die digitale Litfaßsäule</h1>
                 {user !== undefined ? <button id="logout-button" onClick={logout}>Logout</button> : <></>}
-                <div id="nav-links">
-                    <div className="nav-link">{user !== undefined ? <Link to="/administration">Admin Area</Link> : <></>}</div>
-                    <div className="nav-link">{user !== undefined ? <Link to="/add">Organizer Area</Link> : <></>}</div>
-                    <div className="nav-link">{user !== undefined ? <Link to="/approve">Editor Area</Link> : <></>}</div>
-                </div>
+                {areNavLinksVisible && (
+                    <div id="nav-links">
+                        <div className="nav-link">{user !== undefined ?
+                            <Link to="/gallery">Gallery</Link> : <></>}</div>
+                        <div className="nav-link">{user !== undefined ?
+                            <Link to="/administration">Admin Area</Link> : <></>}</div>
+                        <div className="nav-link">{user !== undefined ?
+                            <Link to="/add">Organizer Area</Link> : <></>}</div>
+                        <div className="nav-link">{user !== undefined ?
+                            <Link to="/approve">Editor Area</Link> : <></>}</div>
+                        <div className="nav-link">{user !== undefined ?
+                            <Link to="/categorySelection">Category Selection</Link> : <></>}</div>
+                    </div>
+                )}
             </header>
             <main className="App-main">
                 <Routes>
                     <Route path={"/"} element={<Home login={login} register={register}/>}/>
 
                     <Route element={<ProtectedRoutesUser user={user}/>}>
-                        <Route path="/gallery" element={<EventGallery events={events}/>}/>
+                        <Route path="/gallery" element={<EventGallery user={user} events={events} getAllEvents={getAllEvents} getEventsByCategory={getEventsByCategory}/>}/>
+                        <Route path="/categorySelection"
+                                        element={<PreferredEventCategorySelection user={user} categories={categories}
+                                                                                  updateUserPreferredCategories={updateUserPreferredCategories}/>}/>
                     </Route>
 
                     <Route element={<ProtectedRoutesAdminOnly user={user}/>}>
                         <Route path="/administration" element={<Administration events={events}/>}/>
                     </Route>
 
                     <Route element={<ProtectedRoutesAdminAndOrganizer user={user}/>}>
-                        <Route path="/add" element={<AddEvent events={events}/>}/>
+                        <Route path="/add" element={<AddEvent user={user} events={events}
```

Comment:
Could you simplify this by using an object with these props as props?

Reply:
Do you mean an additional layer? So that props contains only one object, that in turn contains the props i give to AddEvent component at the moment? Not sure. Do you apply this pattern in your project? What would be the benefit in my case?  I wonder, if It might affect the comprehensibility, especially regarding which props are available in the component.